### ISSUE TITLE
LSP Server: Type analysis should consider types that are not explicitely imported

### ISSUE THREAD

**Issue by kbleeke** [untoldwind/KontrolSystem2#107]
Consider the following structure

_test2.to_
```
pub type ResultType = (
    a: int
)

pub fn foo() -> ResultType = {
    (a: 123)
}
```

_test.to_
```
use { foo } from test2

fn bar() -> Unit = {
    let a = foo()
    let b = a.a
}
```

The type for variable `a` is unkown. This can be worked around by importing ResultType as well.

![image](https://github.com/untoldwind/KontrolSystem2/assets/39027073/5e16b65b-ce1d-4dbd-a877-33170f23049f)

However, if the local module defines a type of the same name, the inferred type of `a` becomes `test::ResultType` instead of `test2::ResultType` 

![image](https://github.com/untoldwind/KontrolSystem2/assets/39027073/0b762ef8-4d4b-4147-a23f-d52898bc2044)

**Comment by kbleeke** [comment]
Actually, is seems like the return type of the imported function is already wrong in second case

![image](https://github.com/untoldwind/KontrolSystem2/assets/39027073/d40876f5-a22c-4ce6-9cdd-a9cf0699e4e0)

**Comment by untoldwind** [comment]
Problem was that the type alias `ResultType` was not resolved in the right module context.

**Comment by untoldwind** [comment]
The lsp-server 0.0.12 should handle this correctly now.

**Comment by lefouvert** [comment]
_version 0.5.2.2 (Ckan)_

Hi,
Sorry for my ignorance, but I don't know what is a LSP-server, neither I know if this fix has been published on the version 0.5.2.2 available in Ckan.

However, it also seems to apply to Records elements :

_gitreport/typeanalysis0.to2_
```
pub type FancyConst = int

pub const FancyConstHolder: (Foo: FancyConst, Bar: FancyConst) = (
    Foo: 0,
    Bar: 1
)

pub fn foobar(p: FancyConst) -> Option<bool> = {
    if (p == FancyConstHolder.Foo)
        return Some(true)
    else if (p == FancyConstHolder.Bar)
        return Some(false)
    return None()
}
```

_gitreport/typeanalysis1.to2_
```
use {FancyConstHolder, foobar} from gitreport::typeanalysis0

fn fancyFunc() -> Unit = {
    foobar(FancyConstHolder.Foo)
}
```
give this error, which I think is linked with this topic :
> Rebooted in 00:00:03.9095632
> ERROR: [gitreport\typeanalysis0.to2(4, 35)] InvalidType
Unable to lookup type FancyConst
